klayout/drc/testing/run_regression.py

```diff
@@ -0,0 +1,924 @@
+        rule_counts[rule_name] += 1
+
+    return rule_counts
+
+
+def run_test_case(
+    drc_dir,
+    layout_path,
+    run_dir,
+    table_name,
+):
+    """
+    This function run a single test case using the correct DRC file.
+
+    Parameters
+    ----------
+    drc_dir : string or Path
+        Path to the location where all runsets exist.
+    layout_path : stirng or Path object
+        Path string to the layout of the test pattern we want to test.
+    run_dir : stirng or Path object
+        Path to the location where is the regression run is done.
+    table_name : string
+        Table name that we are running on.
+
+    Returns
+    -------
+    dict
+        A dict with all rule counts
+    """
+
+    # Initial value for counters
+    rule_counts = defaultdict(int)
+
+    # Get switches used for each run
+    sw_file = os.path.join(Path(layout_path.parent).absolute(), f"{table_name}.{SUPPORTED_SW_EXT}")
+
+    if os.path.exists(sw_file):
+        switches = " ".join(get_switches(sw_file, table_name))
+    else:
+        switches = "--variant=C"  # default switch
+
+    # Adding switches for specific runsets
+    if "antenna" in str(layout_path):
+        switches += " --antenna_only"
+    elif "density" in str(layout_path):
+        switches += " --density_only"
+
+    # Creating run folder structure
+    pattern_clean = ".".join(os.path.basename(layout_path).split(".")[:-1])
+    output_loc = f"{run_dir}/{table_name}"
+    pattern_log = f"{output_loc}/{pattern_clean}_drc.log"
+
+    # command to run drc
+    call_str = f"python3 {drc_dir}/run_drc.py --path={layout_path} {switches} --table={table_name} --run_dir={output_loc} --run_mode=flat --thr=1  > {pattern_log} 2>&1"
+
+    # Starting klayout run
+    os.makedirs(output_loc, exist_ok=True)
+    try:
+        check_call(call_str, shell=True)
+    except Exception as e:
+        pattern_results = glob.glob(os.path.join(output_loc, f"{pattern_clean}*.lyrdb"))
+        if len(pattern_results) < 1:
+            logging.error("%s generated an exception: %s" % (pattern_clean, e))
+            traceback.print_exc()
+            raise
+
+    # Checking if run is completed or failed
+    pattern_results = glob.glob(os.path.join(output_loc, f"{pattern_clean}*.lyrdb"))
+
+    # Get list of rules covered in the test case
+    rules_tested = get_unit_test_coverage(layout_path)
+
+    if len(pattern_results) > 0:
+        # db to gds conversion
+        marker_output, runset_analysis = convert_results_db_to_gds(pattern_results[0], rules_tested)
+
+        # Generating merged testcase for violated rules
+        merged_output = generate_merged_testcase(layout_path, marker_output)
+
+        # Generating final db file
+        if os.path.exists(merged_output):
+            final_report = f'{merged_output.split(".")[0]}_final.lyrdb'
+            call_str = f"klayout -b -r {runset_analysis} -rd input={merged_output} -rd report={final_report}"
+            check_call(call_str, shell=True)
```

Comment:
what do we do with the output?

Reply:
Analyze it.

Reply:
Sorry for the lack of clarify, `check_call` doesn't seems to return the standard output or standard error of klayout, does that mean that klayout output will just interleave with the existing logs without being captured?

Reply:
I'll redirect the output to log file.

Reply:
Will be fixed in #7